### PR TITLE
WIP: x/provider: grpc query service

### PR DESCRIPTION
**Pull request by akhilkumarpilli**
ref: #755

### CODE REVIEW

File: proto/akash/market/lease.proto [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+message Lease {
+    option (gogoproto.equal) = false;
+
+    LeaseID lease_id = 1 [
+        (gogoproto.customname) = "LeaseID",
+        (gogoproto.nullable)   = false,
+        (gogoproto.jsontag)    = "id",
+        (gogoproto.moretags)   = "yaml:\"id\""
+    ];
+    uint32 state = 2 [
```
Comment (boz) [inline proto/akash/market/lease.proto:51]: can we use an enum for these?

File: cmd/akashd/genaccounts.go [hunk 2]
```diff
@@ -113,45 +114,57 @@ contain valid denominations. Accounts may optionally be supplied with vesting pa
...
-				return errors.Wrap(err, "failed to unmarshal genesis state")
+				return fmt.Errorf("failed to unmarshal genesis state: %w", err)
 			}
 
-			authGenState := auth.GetGenesisStateFromAppState(cdc, appState)
+			authGenState := authtypes.GetGenesisStateFromAppState(cdc, appState)
 
 			if authGenState.Accounts.Contains(addr) {
-				return errors.Errorf("cannot add account at existing address %s", addr)
+				return fmt.Errorf("cannot add account at existing address %s", addr)
```
Comment (anilCSE) [inline cmd/akashd/genaccounts.go:125]: can we use `sdkerrors` throughout for custom errors?

File: proto/akash/market/lease.proto (hunk 1 above)
Comment (anilCSE) [reply proto/akash/market/lease.proto:51]: Yes, makes sense. Let's use enum for these

File: Makefile [hunk 3]
```diff
@@ -244,4 +244,56 @@ test-sim-after-import:
 	go test -mod=readonly -tags=$(BUILD_MAINNET) $(APP_DIR) -run=TestAppSimulationAfterImport -Enabled=true \
 		-NumBlocks=50 -BlockSize=100 -Commit=true -Seed=99 -Period=5 -v -timeout 10m
 
-test-sims: test-sim-fullapp test-sim-nondeterminism test-sim-import-export test-sim-after-import
\ No newline at end of file
+test-sims: test-sim-fullapp test-sim-nondeterminism test-sim-import-export test-sim-after-import
+
+proto-gen:
+	@./script/protocgen.sh
```
Comment (boz) [inline Makefile:250]: extremely small nitpick: I prefer to see the command that is executing; do you mind not using `@`?

**Pull request merged**